pkg/controller/deployment/default_cert_manager_creator.go

```diff
@@ -0,0 +1,67 @@
+package deployment
+
+import (
+	"context"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/cert-manager-operator/apis/operator/v1alpha1"
+	alpha1 "github.com/openshift/cert-manager-operator/pkg/operator/clientset/versioned/typed/operator/v1alpha1"
+)
+
+type DefaultCertManagerController struct {
+	operatorClient    v1helpers.OperatorClient
+	controllerFactory *factory.Factory
+	recorder          events.Recorder
+	name              string
+	certManagerClient alpha1.OperatorV1alpha1Interface
+}
+
+func (c *DefaultCertManagerController) Name() string {
+	return c.name
+}
+
+func NewDefaultCertManagerController(controllerName string, operatorClient v1helpers.OperatorClient, certManagerClient alpha1.OperatorV1alpha1Interface, eventsRecorder events.Recorder) *DefaultCertManagerController {
+	return &DefaultCertManagerController{
+		operatorClient:    operatorClient,
+		certManagerClient: certManagerClient,
+		controllerFactory: factory.New().ResyncEvery(time.Minute).WithInformers(
+			operatorClient.Informer(),
+		),
+		recorder: eventsRecorder.WithComponentSuffix("default-cert-manager-controller"),
+		name:     controllerName,
+	}
+}
+
+func (c *DefaultCertManagerController) Run(ctx context.Context, workers int) {
+	c.controllerFactory.WithSync(c.Sync).ToController("DefaultCertManager_"+c.name, c.recorder).Run(ctx, workers)
+}
+
+func (c *DefaultCertManagerController) Sync(ctx context.Context, syncCtx factory.SyncContext) error {
```

Comment:
let's not make this method public. the controller factory `Sync` is.

Reply:
Fixed.